Promote Central Slot Status Usage More Obviously

ejoerns opened the pull request:
Using a single central slot status file in RAUC has several advantages. Thus we should be more explicit in notifying the user of this feature.

This now:
* adds a check to the design checlist
* explicitly mentions benefits in documentations
* adds an info message when not using the central status

OnkelUlla commented on docs/reference.rst:
```diff
@@ -98,6 +99,8 @@ Example configuration:
   be stored (e.g. slot specific metadata, see :ref:`slot-status`).
   This file should be located on a filesystem which is not overwritten during
   updates.
+  This method is preferable over per-slots status files as it allows to store
```
To avoid having a recurrence at the beginning of the sentence compared to the previous one I propose to start it like

```
A central status file is preferable…
```

instead. While at it, please do ``s/preferable over per-slots/preferable to per-slots/`` and ``s/per-slots/per-slot/`` as well.

ejoerns replied:
Do you know about GH suggestions? ;) I'll change it as you requested.

OnkelUlla replied:
> Do you know about GH suggestions? ;) I'll change it as you requested.

Having a joint coffee together in the office later will offer a good possibility for introducing me to GH suggestions! ;-)

OnkelUlla commented on docs/reference.rst:
```diff
@@ -98,6 +99,8 @@ Example configuration:
   This file should be located on a filesystem which is not overwritten during
   updates.
+  This method is preferable over per-slots status files as it allows to store
+  data also for read-only or (temporary) filesystem-less slots.
```
Not 100% sure, but for me it feels better to have a replacement ``s/temporary/temporarily/``.

ejoerns replied:
I don't see that one should use an adverb here.

OnkelUlla replied:
Perhaps I got the intended meaning wrong: I thought that it was bound to ``filesystem-less``.

ejoerns commented on docs/reference.rst:
```diff
@@ -27,6 +27,7 @@ Example configuration:
   [system]
   compatible=FooCorp Super BarBazzer
   bootloader=barebox
+  statusfile=/data/global-status.raucs
```
Ok

jluebbe commented:
I'm not sure about adding the fallback warning in the code, in some situations, it might be a good choice to not use a central slot status (no or redundant data partition)/

ejoerns commented:
> I'm not sure about adding the fallback warning in the code, in some situations, it might be a good choice to not use a central slot status (no or redundant data partition)/

It is not a warning indeed. Maybe the next consequence would be to also let the per-slot status file be selected explicitly in config? like:

``` ini
statusfile=per-slot
```

OnkelUlla commented:
> It is not a warning indeed. Maybe the next consequence would be to also let the per-slot status file be selected explicitly in config? like:
> 
> ```ini
> statusfile=per-slot
> ```

This is a good idea as it allows to express that one is aware of central status files and for some reason explicitely does not choose it.

ejoerns commented:
@jluebbe I see this as your agreement on my proposal and will rework it accordingly ;)

ejoerns commented:
I've now changed the PR to support explicitly setting both central and per-slot status storage strategy. I also updated some parts of the documentation and improved the debugability of config_file checks.
